Hunting Queries/SecurityEvent/Least_Common_Process_Command_Lines.txt

```diff
@@ -0,0 +1,18 @@
+// Name: Least Common Processes by Command Line
+// Id: 088d30e9-c02b-46b1-bd1f-d5b6d6b782f0
+// Description: Looks across your environment for least common Process Command Lines, may be noisy and require whitelisting.  By ZanCo
+// DataSource: #SecurityEvent
+// Tactics: #InitialAccess, #Execution, #Persistance
+let Whitelist = dynamic (['conhost.exe', 'foo.exe', 'baz.exe']);
```

Comment:
I tested on a large data set and the perf hit doing it in the Whitelist is almost 3x.  We should exclude conhost.exe right after the initial event lookup instead -
| where EventID == 4688 and TimeGenerated > StartDate and TimeGenerated < (StartDate + Duration)
| where NewProcessName !has "\\conhost.exe"

Reply:
can you try it with endswith instead? that may improve the perf even more.  I just would recommend not getting in the habit of staying away from 20+ lines of OR statements

Reply:
Ah yes, !endswith "\conhost.exe" is better performance here.

Reply:
added conhost endswith clause